Cyclic imports in `simplemma.strategies.dictionaries`

The new `trie_directory_factory` file seems to create cyclic imports as the dictionary strategies import each other (?) – source: pylint.

This is not a problem in itself but could make bugs difficult to find or to trace.

@juanjoDiaz if you want to have a look go ahead.

So for reference this is what `pylint` reports:
```
simplemma/strategies/fallback/lemmatization_fallback_strategy.py:1:0: R0401: Cyclic import (simplemma -> simplemma.lemmatizer -> simplemma.strategies -> simplemma.strategies.dictionaries -> simplemma.strategies.dictionaries.trie_directory_factory) (cyclic-import)
simplemma/strategies/fallback/lemmatization_fallback_strategy.py:1:0: R0401: Cyclic import (simplemma -> simplemma.language_detector -> simplemma.strategies -> simplemma.strategies.dictionaries -> simplemma.strategies.dictionaries.trie_directory_factory) (cyclic-import)
```

I'd probably just break the dependency cycle by moving the definition of the version number to another module like this:

```diff
diff --git a/simplemma/__init__.py b/simplemma/__init__.py
index bf94f83..2bf41cf 100644
--- a/simplemma/__init__.py
+++ b/simplemma/__init__.py
@@ -14,7 +14,6 @@ __title__ = "simplemma"
 __author__ = "Adrien Barbaresi, Juanjo Diaz and contributors"
 __email__ = "barbaresi@bbaw.de"
 __license__ = "MIT"
-__version__ = "1.1.1"
 
 
 from .language_detector import LanguageDetector, in_target_language, langdetect
@@ -26,3 +25,4 @@ from .token_sampler import (
     TokenSampler,
 )
 from .tokenizer import RegexTokenizer, Tokenizer, simple_tokenizer
+from .version import __version__
diff --git a/simplemma/strategies/dictionaries/trie_directory_factory.py b/simplemma/strategies/dictionaries/trie_directory_factory.py
index 8f534ca..8c0c9a3 100644
--- a/simplemma/strategies/dictionaries/trie_directory_factory.py
+++ b/simplemma/strategies/dictionaries/trie_directory_factory.py
@@ -14,7 +14,7 @@ except ImportError:
             raise ImportError("marisa_trie and platformdirs packages not installed")
 
 
-from simplemma import __version__ as SIMPLEMMA_VERSION
+from simplemma.version import __version__ as SIMPLEMMA_VERSION
 from simplemma.strategies.dictionaries.dictionary_factory import (
     DefaultDictionaryFactory,
     DictionaryFactory,
diff --git a/simplemma/version.py b/simplemma/version.py
new file mode 100644
index 0000000..a82b376
--- /dev/null
+++ b/simplemma/version.py
@@ -0,0 +1 @@
+__version__ = "1.1.1"
```

--- a/simplemma/__init__.py
+++ b/simplemma/__init__.py
@@ -10,13 +10,7 @@ Modules:
 
 """
 
-__title__ = "simplemma"
-__author__ = "Adrien Barbaresi, Juanjo Diaz and contributors"
-__email__ = "barbaresi@bbaw.de"
-__license__ = "MIT"
-__version__ = "1.1.1"
-
-
+from .__metadata__ import __title__, __author__, __email__, __license__, __version__
 from .language_detector import LanguageDetector, in_target_language, langdetect
 from .lemmatizer import Lemmatizer, is_known, lemma_iterator, lemmatize, text_lemmatizer
 from .token_sampler import (

--- a/simplemma/__metadata__.py
+++ b/simplemma/__metadata__.py
@@ -1,0 +1,18 @@
+"""
+Package Metadata
+
+This module contains metadata information for the simplemma package.
+
+Attributes:
+    __title__ (str): The title of the package.
+    __author__ (str): The authors of the package.
+    __email__ (str): The contact email for the package.
+    __license__ (str): The license under which the package is distributed.
+    __version__ (str): The current version of the package.
+"""
+
+__title__ = "simplemma"
+__author__ = "Adrien Barbaresi, Juanjo Diaz and contributors"
+__email__ = "barbaresi@bbaw.de"
+__license__ = "MIT"
+__version__ = "1.1.1"

--- a/simplemma/strategies/dictionaries/trie_directory_factory.py
+++ b/simplemma/strategies/dictionaries/trie_directory_factory.py
@@ -14,7 +14,7 @@ except ImportError:
             raise ImportError("marisa_trie and platformdirs packages not installed")
 
 
-from simplemma import __version__ as SIMPLEMMA_VERSION
+from simplemma.__metadata__ import __version__ as SIMPLEMMA_VERSION
 from simplemma.strategies.dictionaries.dictionary_factory import (
     DefaultDictionaryFactory,
     DictionaryFactory,

